src/plugins/discover/public/application/main/hooks/use_discover_state.ts

```diff
@@ -311,13 +320,66 @@ export function useDiscoverState({
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [config, documentState, dataViews]);
 
+  const updateAdHocDataViewId = useCallback(
+    async (dataViewToUpdate: DataView) => {
+      const newDataView = await dataViews.create({ ...dataViewToUpdate.toSpec(), id: undefined });
+
+      dataViews.clearInstanceCache(dataViewToUpdate.id);
+      removeAdHocDataView(dataViewToUpdate.id!);
+      savedSearch.searchSource.setField('index', newDataView);
+
+      // update filters references
+      const uiActions = await getUiActions();
+      const trigger = uiActions.getTrigger(UPDATE_FILTER_REFERENCES_TRIGGER);
+      const action = uiActions.getAction(UPDATE_FILTER_REFERENCES_ACTION);
+      action?.execute({
+        trigger,
+        fromDataView: dataViewToUpdate.id,
+        toDataView: newDataView.id,
+        usedDataViews: [],
+      } as ActionExecutionContext);
+
+      return newDataView;
+    },
+    [dataViews, savedSearch.searchSource]
+  );
+
+  const { openConfirmSavePrompt, updateSavedSearch } = usePersistedDataView(updateAdHocDataViewId);
+  const persistDataView = useCallback(async () => {
+    const currentDataView = savedSearch.searchSource.getField('index')!;
+    if (currentDataView && !currentDataView.isPersisted()) {
+      const createdDataView = await openConfirmSavePrompt(currentDataView);
+      if (createdDataView) {
+        savedSearch.searchSource.setField('index', createdDataView);
+        await onChangeDataView(createdDataView.id!);
+
+        // update saved search with saved data view
+        if (savedSearch.id) {
+          const currentState = stateContainer.appStateContainer.getState();
+          await updateSavedSearch({ savedSearch, dataView: createdDataView, state: currentState });
+        }
+        return createdDataView;
+      }
+      return undefined;
+    }
+    return currentDataView;
+  }, [stateContainer, onChangeDataView, openConfirmSavePrompt, savedSearch, updateSavedSearch]);
+
+  useEffect(() => {
+    if (!dataView.isPersisted() && !savedSearch.id) {
+      getUrlTracker().setTrackedUrl('/');
```

Comment:
This works just at the initial selection of the ad hoc data view selection, but url is still tracked when there are changes to query ... I think the best solution would be to solve the issue Joe encountered ... having a closer look currently